components/tikv_util/src/macros.rs

```diff
@@ -171,6 +171,29 @@ macro_rules! try_opt_or {
     }};
 }
 
+/// It checks whether the current thread is unwinding because of panic. If it is,
+/// log an error message instead of causing double panic.
```

Comment:
Oh, @sticnarf can you mention that this is strictly for use in `drop`?

Reply:
I don't want to say that because it's ok to use this macro as a general replacement of `panic!`. But I modify the doc telling why we need this macro. Please take a look again. @brson